exchanges/stream/websocket.go

```diff
@@ -259,7 +259,7 @@ func (w *Websocket) SetupNewConnection(c ConnectionSetup) error {
 
 // Connect initiates a websocket connection by using a package defined connection
 // function
-func (w *Websocket) Connect() error {
+func (w *Websocket) Connect(subscriptionModifier ...subscription.Hook) error {
```

Comment:
Yeah I see your issues;

I kept it a vague function hook so that it doesn't impact gct in terms of line changes and it's optional, and it makes it more obvious using it from a library perspective. 

I agree, connect shouldn't handle subs, so the open multiconnection PR's will require subs to perform a connection in the first place 0 subs 0 connections. So that might be problematic to the design you are currently using when that is merged. 

> You can likely hook in at a different point if things are split out more. So that your order of operations in your application is more like: ConnectWithoutSubscribing()->subs := GenerateSubscriptions()>supercoolapp.ModifySubscriptionsToWhatIWantPleaseDontPanic(subs)>Subscribe(subs)

Having that functionality exposed it may take more time to flesh out, if you have something you are working towards, if you can cherry-pick that across and PR that would be helpful. 

I am opting to close this and I will vendor this change in my external works until I have a more gct friendly version(if ever), so as to not waste time with design and implementation, then review process, it doesn't seem worth it for now for what it is trying to achieve.